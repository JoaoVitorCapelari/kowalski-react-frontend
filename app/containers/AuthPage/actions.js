/*
 *
 * AuthPage actions
 *
 */

import {
  DEFAULT_ACTION,
  CHANGED_PASSWORD,
  CHANGED_USERNAME,
  SUBMITTED_CREDENTIALS,
  SERVER_RESPONDED,
} from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}

export function changedUsername(username) {
  return {
    type: CHANGED_USERNAME,
    payload: username,
  };
}

export function changedPassword(password) {
  return {
    type: CHANGED_PASSWORD,
    payload: password,
  };
}

export function submitCredentials({ username, password }) {
  return {
    type: SUBMITTED_CREDENTIALS,
    payload: { username, password },
  };
}

export function serverResponded(payload) {
  return {
    type: SERVER_RESPONDED,
    payload,
  };
}
