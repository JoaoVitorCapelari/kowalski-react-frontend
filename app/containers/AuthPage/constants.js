/*
 *
 * AuthPage constants
 *
 */

export const DEFAULT_ACTION = 'app/AuthPage/DEFAULT_ACTION';
export const CHANGED_USERNAME = 'app/AuthPage/CHANGED_USERNAME';
export const CHANGED_PASSWORD = 'app/AuthPage/CHANGED_PASSWORD';
export const SUBMITTED_CREDENTIALS = 'app/AuthPage/SUBMITTED_CREDENTIALS';
export const SERVER_RESPONDED = 'app/AuthPage/SEVER_RESPONDED';
