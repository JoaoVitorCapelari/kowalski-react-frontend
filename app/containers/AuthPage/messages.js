/*
 * AuthPage Messages
 *
 * This contains all the text for the AuthPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  friendlyIntroductoryMessage: { id: 'app.containers.AuthPage.friendlyIntroductoryMessage' },
  username: { id: 'app.containers.AuthPage.LoginForm.username' },
  password: { id: 'app.containers.AuthPage.LoginForm.password' },
  loginButton: { id: 'app.containers.AuthPage.LoginForm.loginButton' },
  signinButton: { id: 'app.containers.AuthPage.signInButton' },
  signupButton: { id: 'app.containers.AuthPage.signUpButton' },
  appTitle: { id: 'app.title' },
  pageTitle: { id: 'app.containers.AuthPage.pageTitle' },
  pageDescription: { id: 'app.containers.AuthPage.pageDescription' },
  version: {
    id: 'app.containers.AuthPage.version',
    values: {
      version: process.env.SATURNO_VERSION,
    },
  },
});
