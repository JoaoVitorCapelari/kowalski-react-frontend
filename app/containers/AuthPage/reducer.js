/*
 *
 * AuthPage reducer
 *
 */

import { fromJS } from 'immutable';
import {
  DEFAULT_ACTION,
  CHANGED_USERNAME,
  CHANGED_PASSWORD,
  SUBMITTED_CREDENTIALS,
  SERVER_RESPONDED,
} from './constants';

const initialState = fromJS({
  username: '',
  password: '',
  isSubmitting: false,
  serverErrorMsg: '',
});

function authPageReducer(state = initialState, action) {
  switch (action.type) {
    case DEFAULT_ACTION:
      return state;
    case CHANGED_USERNAME:
      return state.set('username', action.payload);
    case CHANGED_PASSWORD:
      return state.set('password', action.payload);
    case SUBMITTED_CREDENTIALS:
      return state
      .set('isSubmitting', true);
    case SERVER_RESPONDED:
      return state
        .set('isSubmitting', false)
        .set('serverErrorMsg', action.payload.success ? '' : action.payload.errorMsg);
    default:
      return state;
  }
}

export default authPageReducer;
