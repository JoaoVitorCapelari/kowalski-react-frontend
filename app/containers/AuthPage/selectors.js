import { createSelector } from 'reselect';
import { fromJS } from 'immutable';

/**
 * Direct selector to the authPage state domain
 */
const selectAuthPageDomain = (state) => state.get('authpage') || fromJS({});

/**
 * Other specific selectors
 */


/**
 * Default selector used by AuthPage
 */

const makeSelectAuthPage = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.toJS()
);

const makeSelectUsername = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.get('username'),
);

const makeSelectPassword = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.get('password'),
);

const makeSelectIsSubmitting = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.get('isSubmitting'),
);

const makeSelectServerErrorMsg = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.get('serverErrorMsg'),
);

export default makeSelectAuthPage;
export {
  selectAuthPageDomain,
  makeSelectUsername,
  makeSelectPassword,
  makeSelectIsSubmitting,
  makeSelectServerErrorMsg,
};
