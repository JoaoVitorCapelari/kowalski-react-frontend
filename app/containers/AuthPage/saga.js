import { takeEvery, put, call } from 'redux-saga/effects';
import { push } from 'react-router-redux';
import { authenticate } from '../../support/backend/KowalskiBackendClient';

import {
  SUBMITTED_CREDENTIALS,
} from './constants';

import {
  serverResponded,
} from './actions';

import { SERVER_BASE_URL } from '../../utils/constants';

export function* handleSubmission({ payload: { username, password } }) {
  try {
    const serverResponse = yield call(authenticate, { config: { baseUrl: SERVER_BASE_URL }, username, password });
    const token = serverResponse.headers.get('Authorization').match(/Bearer (.*)/)[1];
    localStorage.setItem('authToken', token);
    yield put(serverResponded({ success: true }));
    yield put(push('/'));
  } catch (error) {
    yield put(serverResponded({ success: false, errorMsg: 'Invalid credentials' }));
  }
}

// Individual exports for testing
export default function* defaultSaga() {
  yield takeEvery(SUBMITTED_CREDENTIALS, handleSubmission);
}
