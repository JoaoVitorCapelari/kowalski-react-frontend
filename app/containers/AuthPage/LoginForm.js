/**
 *
 * AuthPage
 *
 */

import React from 'react';
// import { connect } from 'react-redux';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
// import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import styled from 'styled-components';

// import injectSaga from 'utils/injectSaga';
// import injectReducer from 'utils/injectReducer';
// import makeSelectAuthPage from './selectors';
// import reducer from './reducer';
// import saga from './saga';
import messages from './messages';

const ButtonWrapper = styled.button`
  width: 100%
`;

function AuthPage({ intl }) {
  return (
    <form>
      <div className="field">
        <div className="control">
          <input type="text" className="input" placeholder={intl.formatMessage(messages.username)} />
        </div>
      </div>
      <div className="field">
        <div className="control">
          <input type="password" className="input" />
        </div>
      </div>
      <div className="control">
        <ButtonWrapper type="submit" className="button is-primary">
          <FormattedMessage {... messages.loginButton} />
        </ButtonWrapper>
      </div>
    </form>
  );
}

AuthPage.propTypes = {
  intl: intlShape.isRequired,
};

// const mapStateToProps = createStructuredSelector({
//   authpage: makeSelectAuthPage(),
// });

// function mapDispatchToProps(dispatch) {
//   return {
//     dispatch,
//   };
// }

// const withConnect = connect(mapStateToProps, mapDispatchToProps);

// const withReducer = injectReducer({ key: 'authpage', reducer });
// const withSaga = injectSaga({ key: 'authpage', saga });

export default compose(
  // withReducer,
  // withSaga,
  // withConnect,
)(injectIntl(AuthPage));
