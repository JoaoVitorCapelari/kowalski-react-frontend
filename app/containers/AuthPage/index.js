/**
 *
 * AuthPage
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectAuthPage from './selectors';
import reducer from './reducer';
import saga from './saga';
import messages from './messages';

function AuthPage() {
  return (
    <div>
      <Helmet>
        <title>AuthPage</title>
        <meta name="description" content="Description of AuthPage" />
      </Helmet>
      <FormattedMessage {...messages.header} />
    </div>
  );
}

AuthPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  authpage: makeSelectAuthPage(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: 'authpage', reducer });
const withSaga = injectSaga({ key: 'authpage', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(AuthPage);
