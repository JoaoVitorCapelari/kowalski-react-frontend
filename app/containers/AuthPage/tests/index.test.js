import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { IntlProvider } from 'react-intl';
import createHistory from 'history/createMemoryHistory';

import AuthPage from '../index';
import { translationMessages } from '../../../i18n';
import configureStore from '../../../configureStore';

const initialState = {};
const history = createHistory();
const store = configureStore(initialState, history);

describe('<AuthPage />', () => {
  it('renders', () => {
    const renderedComponent = mount(
      <Provider store={store}>
        <IntlProvider locale="en" messages={translationMessages.en}>
          <AuthPage />
        </IntlProvider>
      </Provider>
    );
    expect(renderedComponent.html()).toMatchSnapshot();
  });
});
