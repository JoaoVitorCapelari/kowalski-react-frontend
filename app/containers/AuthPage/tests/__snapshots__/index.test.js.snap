// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthPage /> renders 1`] = `"<div class=\\"columns AuthPage__ColumnsWrapper-dIFxNP jbFEVe\\"><!-- react-empty: 2 --><div class=\\"column has-text-centered auth-page-left-container\\"><h3 class=\\"AuthPage__H3Wrapper-cZQfkt lolVTS\\"><span>Your best friend to help in daily management tasks.</span></h3><img src=\\"IMAGE_MOCK\\" alt=\\"Illustration of a calendar\\"><div class=\\"AuthPage__VersionContainer-gfNiAV hzdMTa\\"><span>Version </span></div></div><div class=\\"column auth-page-right-container AuthPage__RightPanelWrapper-jCFRuh inAMxh\\"><div class=\\"AuthPage__FormTitleWrapper-diEAXu ePtDmx\\"><span class=\\"kowalski-primarily-colored\\"><!-- react-text: 12 --> <!-- /react-text --><img src=\\"IMAGE_MOCK\\" alt=\\"Saturno - Logo\\"></span></div><div class=\\"AuthPage__FormWrapper-bdCXen jYKpgi\\"><form><div class=\\"field\\"><div class=\\"control\\"><input type=\\"text\\" class=\\"input\\" placeholder=\\"Username\\" value=\\"\\"></div></div><div class=\\"field\\"><div class=\\"control\\"><input type=\\"password\\" class=\\"input\\" value=\\"\\"></div></div><div class=\\"control\\"><button class=\\"button is-primary  LoginForm__ButtonWrapper-QBjvv bVNTI\\" type=\\"submit\\"><span>LOGIN</span></button></div><!-- react-text: 25 --><!-- /react-text --></form></div></div></div>"`;
